include/utils.h

```diff
@@ -1,14 +1,19 @@
-#include "affinity.h"
-#include "logger.h"
+#pragma once
+
+#include <type_traits>
 
 namespace celerity {
 namespace detail {
```

Comment:
I don't think `popcount` should be in `detail`, it's not an implementation detail. Maybe it should be in a `utils` namespace? Or just in the `celerity` namespace.

@psalz Does celerity not have a place for utility functions, is this really the first utility function for celerity?

Reply:
> I don't think popcount should be in detail, it's not an implementation detail. Maybe it should be in a utils namespace? Or just in the celerity namespace.

That's actually fine, imo. Everything in the `celerity` namespace is part of the public (user-facing) API, everything else lives inside `detail`. We could have a `detail::utils` namespace though - no strong opinion there.

> @psalz Does celerity not have a place for utility functions, is this really the first utility function for celerity?

Nope, [this](https://github.com/celerity/celerity-runtime/blob/da1c240e0146b32f1fa095346f5bd99a18ca3bf2/include/graph_generator.h#L24) [was](https://github.com/celerity/celerity-runtime/blob/da1c240e0146b32f1fa095346f5bd99a18ca3bf2/include/intrusive_graph.h#L18) the approach taken so far :P.

Reply:
Right, `popcount` is not something that celerity provides. In which case I would be in favor of putting it inside `detail::utils` and having this header be the `utils` header where `TODO move to utils` code can be refactored to in the future.